force-app/main/default/classes/AddNewProductCustomController.cls

```
@@ -0,0 +1,164 @@
+                PricebookEntry pbe = pbeMap.get(p.Id) ?? null;
+                ProductWrapper pw = new ProductWrapper();
+                pw.pricebookEntryId = pbe?.Id;
+                pw.unitPrice = pbe?.UnitPrice;
+                pw.productUrl = '/lightning/r/Product2/' + p.Id + '/view';
+                pw.Id = p.Id;
+                pw.productId = p.Id;
+                pw.name = p.Name;
+                pw.productCode = p.ProductCode;
+                pw.description = p.Description;
+                pw.totalQuantity = p.Total_Quantity__c;
+                pw.onHoldQuantity = p.On_Hold_Quantity__c;
+                pw.totalPrice = 0;
+                productWrappers.add(pw);
+            }
+            opw.productList = productWrappers;
+        }
+        else {
+            opw.stdPbId = [Select Id from PriceBook2 Where IsStandard = true AND IsActive = true limit 1] ?.Id;
+        }
+        return JSON.serialize(opw);
+    }
+
+    /**
+        * @description: Modifying the line items through the datatable and save them in the DB.
+    **/
+    @AuraEnabled(cacheable=false)
+    public static String saveProducts(String recordData, String recId) {
+        List<ProductWrapper> wc = (List<ProductWrapper>)JSON.deserialize(recordData, List<ProductWrapper>.class);
+        List<OpportunityLineItem> lstOpp = new List<OpportunityLineItem>();
+
+        for (ProductWrapper pw : wc) {
+            OpportunityLineItem oli = new OpportunityLineItem();
+            oli.Quantity = pw.quantity;
+            oli.UnitPrice = pw.unitPrice;
+            oli.ServiceDate = pw.selectedDate;
+            oli.Description = pw.lineDescription;
+            oli.OpportunityId = recId;
+            oli.Product2Id = pw.productId;
+            oli.PricebookEntryId = pw.pricebookEntryId;
+            lstOpp.add(oli);
+        }
+
+        try {
+            insert lstOpp;
+            return 'success';
+        } catch (Exception e) {
+            System.debug(e.getMessage());
```

Throw the exception instead of return.

---

addressed the throw exception instead of return.